src/ORM/Association/HasMany.php

```diff
@@ -392,11 +392,15 @@ public function replace(EntityInterface $sourceEntity, array $targetEntities, ar
      */
     protected function _unlinkAssociated(array $properties, EntityInterface $entity, Table $target, array $remainingEntities = [], array $options = [])
     {
-        $primaryKey = (array)$target->primaryKey();
         $exclusions = new Collection($remainingEntities);
         $exclusions = $exclusions->map(
-            function ($ent) use ($primaryKey) {
-                return $ent->extract($primaryKey);
+            function ($ent) use ($target) {
+                $fields = $ent->extract((array)$target->primaryKey());
```

Comment:
Is there a need to call `primaryKey()` N times? Seems like once would be enough.